eth/abc.py

```diff
@@ -334,10 +335,53 @@ def is_genesis(self) -> bool:
         ...
 
 
+class WitnessIndexAPI(ABC):
+    @property
+    @abstractmethod
+    def hashes(self) -> Tuple[Hash32, ...]:
+        ...
+
+    @property
+    @abstractmethod
+    def accounts_queried(self) -> Collection[Address]:
```

Comment:
*nitpick* 

Why the loose return type?  Typically we are strict about return types (requiring this be something concrete like `Tuple/List`.  Maybe worth documenting why it's allowed to be loose if there is a good reason.

Reply:
Right now it returns a dictionary's key view. I think I just didn't know the right `typing` class to use for that one. But... it's probably a bad idea to blindly return the key view anyway, since you can get a `RuntimeError` while iterating. So I'll just copy it to a tuple or something and make this type more specific.